### ISSUE TITLE
Incorrect DID returned for domain verification

### ISSUE THREAD

**Issue by inedyalkov** [bluesky-social/social-app#6694]
### Steps to Reproduce

1. Navigate to the Bluesky handle setup page.
2. Enter a custom domain name for the handle (e.g., @mydomain.com).
3. Add the DNS record as instructed by Bluesky with the correct DID.
4. Click on Verify DNS record in Bluesky.
5. Observe the error message: "Wrong DID returned from server."

### Attachments

![Screenshot 2024-11-24 at 17 48 28](https://github.com/user-attachments/assets/aefd0155-e26e-4dc4-9bc5-02bf36de9329)

### What platform(s) does this occur on?

iOS, Web (Desktop)

### What version of the app are you using?

1.94

### Additional Information

I have verified the DNS records independently using Google Admin Toolbox Dig, and the correct DID is returned from the DNS server.

**Comment by twhitney11** [comment]
I'm having this same issue. Been messing with it for the past few days. I am running my own PDS and thus want to create a new account with my custom domain handle after setting it up. Changed the DNS TXT record for _atproto, but the app is still reporting the old one. Ironically, I went the .well-known route with the same results, reporting my old DID even though I never had a file in the .well-known before with my old DID. 

Is there a cache somewhere that keeps them, perhaps?

**Comment by tiferrei** [comment]
Same issue here. Contacted BlueSky support and I was told "We forwarded this for a DID flushing". However unfortunately no change so far (~10 days ago).

**Comment by iRyanBell** [comment]
> Same issue here. Contacted BlueSky support and I was told "We forwarded this for a DID flushing". However unfortunately no change so far (~10 days ago).

Running into the same issue here. I have the updated value set on my domain's DNS and uploaded for HTTP verification (plus I've already done email verification for the domain) but it's failing, returning an old value. They must cache this instead of checking latest, as both methods fail.

I emailed support, no answer. Such decentralization.

**Comment by twhitney11** [comment]
![Uploading Screenshot 2024-12-03 at 12.22.16 AM.png…]()

**Comment by twhitney11** [comment]
Still no change, reporting my old DID, even though my DNS is definitely updated. Its giving the same error when using the file in the .well-known directory even though there was never a file there until I put this brand new DID there, so the one its reporting was never in that file. Definitely a bug, I'm having troubles finding where this verification is handled in the codebase. But I did just open a request via the web app's Zendesk.

**Comment by Tsnstuff** [comment]
Exact same issue here. I had verified my domain with another account which I deleted when I created my own pds. I couldn’t figure out how to just move my old one to the pds. It was a new account so I didn’t really research much anyway. 

However for the new account I’d love to be able to use my domain and it is stuck on the old did info.

I’ve tried everything, including the file, which just like someone else posted, never existed with the old did info. It keeps telling me that it finds the old info and thus can’t verify. 

I also got the email from support letting me know they were escalating but I haven’t gotten anything since. 

I’m convinced that somewhere, the system has simply linked my domain with that old did, and now it simply looks at that, it is not looking at the file or the dns txt record. 

I did also try misspelling my domain to see if it would just give me the same error anyway, and no, it gave me the expected error for a domain with no such record.

**Comment by jordijc** [comment]
I'm having exactly the same error. Impossible to validate my own domain as a handle. I used the tool to debug DNS settings and it passes perfectly. But when trying to configure on Bluesky, I get the **Wrong DID returned from server** message. 

Using 
Debugging my handle -> [https://bsky-debug.app/handle?handle=retallsdecuina.cat](https://bsky-debug.app/handle?handle=retallsdecuina.cat)
Attached both screenshots, the PASS debug test. And the Wrong DID message.
Hope this could be solved and we can all use our ow domains as a personal handle.

![change-handle](https://github.com/user-attachments/assets/735d5526-c99e-4f79-b5ba-111067ef42bb)
![handle-check](https://github.com/user-attachments/assets/85f03bae-6071-4b7d-9003-1b178de1df4d)

**Comment by ebeldner** [comment]
### SOLUTION (at least for one flavor of this error): 

I was getting the same error message and was able to solve it. (Part of the challenge is that Bluesky's error messaging seems pretty imprecise; it makes troubleshooting hard.)

My issue turned out to be that my server wasn't returning the expected `Content-Type: text/plain` header: 

> The expected payload is a DID (such as did:plc:abcdef...) with content-type text/plain. The handle resolution function explicitly expects a HTTP 200 OK status. (from [How to verify your Bluesky account](https://bsky.social/about/blog/4-28-2023-domain-handle-tutorial))

In my case, Bsky was returning the correct DID, which was even more confusing: 

<img width="611" alt="Screenshot 2024-12-11 at 11 39 22" src="https://github.com/user-attachments/assets/309843db-bd8b-4a77-9ad4-30a1e6bbe05a" />

Because it was clearly returning the correct information and I could see successful requests for the file at the URL it didn't seem like a problem with DNS or the mod_rewrite rules I needed to get it to work in my hosting environment. So I guessed if it might be some sort of encoding type issue & remembered the `Content-Type` requirement.  

To solve: 
1. I used a `curl -v` command to review my headers. I confirmed there was no Content-Type header being sent. 
2. I modified the .htaccess file for the ./well-known/ directory to include 
```
<Files "atproto-did">
    ForceType text/plain
</Files>
```

Handle is now verified!

**Comment by mu-lambda** [comment]
Have exact same issue

> SOLVED for me!
> 
> I opened a support ticket and they solved it for me. Them made a manual reset on my account to delete the old DID and let me to verify my own domain.
> 
> Handle verified and working!

Have exact same issue, where did you open a support ticket? "Feedback" for bsky.social?

**Comment by inedyalkov** [comment]
> SOLVED for me!
> 
> 
> 
> I opened a support ticket and they solved it for me. Them made a manual reset on my account to delete the old DID and let me to verify my own domain.
> 
> 
> 
> Handle verified and working! 

If only! I'm still waiting for their response.

**Comment by Jaczel** [comment]
> Exact same issue here. I had verified my domain with another account which I deleted when I created my own pds. I couldn’t figure out how to just move my old one to the pds. It was a new account so I didn’t really research much anyway.
> 
> However for the new account I’d love to be able to use my domain and it is stuck on the old did info.
> 
> I’ve tried everything, including the file, which just like someone else posted, never existed with the old did info. It keeps telling me that it finds the old info and thus can’t verify.
> 
> I also got the email from support letting me know they were escalating but I haven’t gotten anything since.
> 
> I’m convinced that somewhere, the system has simply linked my domain with that old did, and now it simply looks at that, it is not looking at the file or the dns txt record.

I had the same issue. Set up a PDS using my domain. Created an account on there and linked my old DID to that. I have since deleted the PDS. The DID will not update on bsky.app and only shows the old DID despite having updated the TXT record hours ago.
I will submit a support ticket as well.

**Comment by twhitney11** [comment]
Hey folks, just an update from me. For what its worth. I did make sure the headers were correctly being sent. They were, I use NGINX so I tried both with an actual file and also just returning the DID direct via NGINX config. Either way, no luck. I opened a support request. It took a LONG time (2 weeks) to get a reply. I got a reply indicating they were escalating the issue to the engineering team to clear the cache for the DID, that was over a week ago. So definitely progress, just slow. I'll let you know when I get it resolved from the BlueSky team.

I thought perhaps I could find in the codebase where this was caching was handled, but sounds like its more of a system admin task than a software development task. From a software standpoint, I still think there must be an easier way or something going on, but I haven't had the time to check the codebase to see how that cache is handled, if its even in there.

**Comment by Tsnstuff** [comment]
I'd like to provide an update that they did manually reset my DID info and I was finally able to link my Domain to my account.  

This process however took a very long time.  I don't remember when exactly I contacted them originally but they did get back to me on December 4th letting me know they escalated the email to manually reset the DiD.  

December 9th I got an email letting me know they had manually reset the DiD.  However, I did everything they told me to do and it still didn't work, I was still getting the old DID.  

December 9th I emailed them back the same day to let them know it didn't work. 

December 18th I had not gotten a reply yet, so I asked for an update. 

December 20th, still nothing from them, So I asked for an update again. 

December 25th, still nothing, so I asked for an update again. 

January 2nd I get the following reply: 

"Apologies for the delay in our response.
 
We have reached out to the relevant team to follow-up on the DID flush. Rest assured that we are looking into this, and we'll get back to you as soon as we have an update."

January 7th I get another email letting me know that my DID was manually refreshed.  This time it finally worked. 

I'm not a developer so I don't really know how much work it would be, maybe I'm being unrealistic, but I would assume the system would go look up the DID information in the actual DNS entries every time rather than be looking at some cache.  This whole experience kind of soured Bluesky to me.

**Comment by twhitney11** [comment]
Mine is fixed now too, similar experience to yours. I didn't reach out for updates consistently, I think only once. But after a long wait (few weeks) I got the message they were escalating. Then another few weeks and I got the message that it had been fixed. I agree it should immediately look at the file in .well-known or immediately do a DNS lookup and NOT use some cache. Still wish I knew if this was a part of the application code somewhere we could see or if its some proprietary thing on the Bsky side?

**Comment by Lach-Arkwight** [comment]
A friend of mine who uses @42NA.girlfailure.zone is also having the issue. It's still pointing to her old DID on Bluesky, but the DNS record on Cloudflare has already changed to her new DID.

**Comment by Lach-Arkwight** [comment]
> A friend of mine who uses @42NA.girlfailure.zone is also having the issue. It's still pointing to her old DID on Bluesky, but the DNS record on Cloudflare has already changed to her new DID.

Ah, nevermind. I was able to get it updated for them through a small script located here: https://gist.github.com/IMB11/b215dfc92d827c03d05f4981308cfd12

**Comment by thombles** [comment]
Quick update, the issue appears to still be in play. I raised a cheerful support request through my account one week ago with screenshots of the error and DNS setup and they got back to me today that it's fixed (and it is). I had tried using `.well-known` to work around it but it seems the cached DNS record took precedence and it didn't help.

**Comment by TechRemarker** [comment]
Same issue. I had previously associated a domain with my account, but needed to change it. When using the DNS method with CloudFlare, it said it was returning the wrong one, even though correct one was there and external checkers all showed it was correct. So appears to be Bsky caches the original one and never actually rechecks, unless presumably support manually intervenes to refresh on their end. Since people said weeks for response/resolution, decided to try the file method. 

Tried that and it didn't work. Saw people mentioned you have to set to type plain. SSH'ed into DigitalOcean which uses NGINX was able to do that. Still didn't work. After more debugging I tried turning off CloudFlare proxy for the A record in DNS then tried again and worked!

So hopefully they fix the caching bug for DNS, but in the meantime, setting the type to plain does appear to resolve it, so long as if you use CloudFlare, that you temporarily disable the proxy (orange cloud) protection.
